includes/subscriptions/class-wc-payments-subscriptions-migrator.php

```diff
@@ -0,0 +1,260 @@
+		 *
+		 * To ensure that all migrated subscriptions have a pending scheduled action, we need to reschedule the next payment date by
+		 * updating the date on the subscription.
+		 */
+		if ( $subscription->has_status( 'active' ) && $subscription->get_time( 'next_payment' ) > time() ) {
+			$new_next_payment = gmdate( 'Y-m-d H:i:s', $subscription->get_time( 'next_payment' ) + 1 );
+			$subscription->update_dates( [ 'next_payment' => $new_next_payment ] );
+
+			$this->log( sprintf( '---- Next payment date updated to %s to ensure active subscription has a pending scheduled payment.', $new_next_payment ) );
+		}
+
+		$this->log( '---- SUCCESS: Subscription migrated.' );
+	}
+
+	/**
+	 * Fetches and validates the subscription to migrate. Returns false if the subscription cannot be migrated.
+	 *
+	 * Only allows migration if:
+	 * - The WooCommerce Subscription extension is active
+	 * - Store is not in staging mode or is a duplicate site
+	 * - Subscription ID belongs to a valid subscription
+	 *
+	 * @param int $subscription_id The ID of the subscription to migrate.
+	 *
+	 * @return WC_Subscription|bool
+	 */
+	private function fetch_and_validate_subscription_to_migrate( $subscription_id ) {
+		if ( ! class_exists( 'WC_Subscriptions' ) ) {
+			$this->log( sprintf( 'Skipping migration of subscription #%d. The WooCommerce Subscriptions extension is not active.', $subscription_id ) );
+			return false;
+		}
+
+		if ( WC_Payments_Subscriptions::is_duplicate_site() ) {
+			$this->log( sprintf( 'Skipping migration of subscription #%d. Site is in staging mode.', $subscription_id ) );
+			return false;
+		}
+
+		$subscription = wcs_get_subscription( $subscription_id );
+		if ( ! $subscription ) {
+			$this->log( sprintf( 'Skipping migration of subscription #%d. Subscription not found.', $subscription_id ) );
+			return false;
+		}
+
+		return $subscription;
+	}
+
+	/**
+	 * Checks if the subscription has already been migrated.
+	 *
+	 * Looks for the `_migrated_wcpay_subscription_id` metadata  on the subscription and if found,
+	 * this function double-checks the subscription in Stripe is not active.
+	 *
+	 * @param WC_Subscription $subscription The WC subscription to migrate.
+	 *
+	 * @return bool
+	 */
+	private function has_subscription_already_migrated( $subscription ) {
+		$migrated_wcpay_subscription_id = $subscription->get_meta( '_migrated_wcpay_subscription_id', true );
+
+		if ( empty( $migrated_wcpay_subscription_id ) ) {
+			return false;
+		}
+
+		try {
+			$wcpay_subscription = WC_Payments::get_payments_api_client()->get_subscription( $migrated_wcpay_subscription_id );
+		} catch ( API_Exception $e ) {
+			$this->log( sprintf( 'Error fetching an already migrated subscription (%s). Could not verify the subscription is canceled in Stripe. %s', $migrated_wcpay_subscription_id, $e->getMessage() ) );
+		}
+
+		if ( ! empty( $wcpay_subscription['status'] ) ) {
```

Comment:
I've changed how we handle migrating subscriptions that are marked as already migrated.
If it's migrated, we just skip the migration with a message:
```
Skipping migration of subscription #231 (sub_XXXX). Subscription has already been migrated.
```

Having this code in the migration was overly complicating things and it wasn't really serving a purpose because there shouldn't be any need to cancel a subscription at Stripe again after we have meta on the subscription to suggest it's been migrated and canceled.

We can revisit how we handle already migrated subscriptions later if it becomes a problem